### PR TITLE
Replace use of `has_version` with appropriate `python_has_version` in `python_check_deps()`

### PR DESCRIPTION
**Pull request by Kangie**
This PR addresses an issue that I encountered cross-compiling `kde-framework/breeze-icons` where an incorrect `has_version` call fails to correctly locate python dependencies resulting in a failed build.

Looking at the guide `python_has_version` should be preferred if the `python-any-r1` eclass is in use. ~~If `python-r1` is in use instead I've just tidied up usage of `has_version`.~~ Fun fact: I just learnt that `python_has_version` comes from `python-utils-r1`, so I'll go back and amend any commits where has_version is used in `python_check_deps()`.

I've also taken the time to do any required cleanup since I'm already looking at....200 odd ebuilds.

- EAPI bumped where trivial
- python 3.11 added to `PYTHON_COMPAT`
- unused / should not have been inherited eclasses removed

Packages were identified by running `grep -rsinl python_gen_any | xargs grep -L "has_version -b"` over ::gentoo.

For @thesamesam

### CODE REVIEW

**Comment by Kangie** [comment]
I figured that it was time to make the QA bot unhappy.

I've got a good few packages / ebuilds / versions to go through, and while I've compiled a good number of these (particularly with more complex `python_check_deps()` functions) I can't be sure that nothing is broken yet, so please do not merge.

**Comment by Kangie** [comment]
Also, noting that `python_has_version` comes from the `python-utils-r1`  eclass, and is

> A convenience wrapper for has_version() with verbose output and better defaults for use in python_check_deps().

I'd like to suggest looking into enabling a QA rule for using `python_has_version` in `python_check_deps()` (or the inverse, no `has_version` in there.)

It would be interesting to look into amending python_has_deps so that it can accept an array to make handling use flags easier, but one thing at a time.

**Review by mgorny — CHANGES_REQUESTED** [review]
Thanks for doing this. I took only a quick look at your latest changes but a few general suggestions:

1. Doing this many packages in one PR is, well, problematic to merge. You really should start small, ideally choose a small subset with common maintainer (ideally @gentoo/python to ease initial review) and start from there.
2. The commit message about "fixing" suggests that something is broken. It's not — while making `python_has_version` makes a lot of sense, let's call it "use python_has_version" or alike.
3. As for the policy, I suppose having pkgcheck for it makes a lot of sense. CC @thesamesam, @arthurzam, I think they've played with tree-sitter magic in pkgcheck.
4. Bumping EAPI is certainly worthwhile but makes things harder to test. You may want to do that separately, and ideally revbump while doing it.
5. For updating copyright headers, you may use `copybump` from mgorny-dev-scripts. Basically `copybump $(git diff --name-only origin)` to deal with all the ebuilds you've changed.
6. `git-fixup` from the same package (or `git commit --fixup=<COMMIT-ID>` to conveniently add fixups for individual commits. Then `git rebase -i` will automatically put them in the right place.

File: x11-wm/mutter/mutter-42.1.ebuild
Comment (thesamesam) [inline x11-wm/mutter/mutter-42.1.ebuild:5]: Just to check: you ran test suites for all of these too to make sure they pass, if adding 3.11? If you did, then great!

Comment (Kangie) [reply x11-wm/mutter/mutter-42.1.ebuild:5]: Not all (or admittedly, most) yet, but I also don't expect this PR to be merged, just want to catch any immediate issues in review before my branch gets too far behind.

As I break it up over the next few days I'll properly test packages and fix or revert anything that doesn't work.

**Comment by thesamesam** [comment]
Would you mind taking this PR and say, splicing it into the python@ maintained packages so we can get some of it in? I can handle the other nits.

**Pull request closed without merging**
